### DISCUSSION TITLE
HmIP-RFUSB wird im Add-On unter "HomeAssistant Supervised" nicht erkannt

### DISCUSSION THREAD

**Discussion by cfkone** [OpenCCU/OpenCCU#1931] [🇩🇪 Fragen&Antworten]
Hallo zusammen,

vorab: ich bin neu in diesem Forum und hoffe, dass ich die richtige Discussion für mein Problem erwischt habe.
Alle ähnlichen Threads führten mich nicht zur Lösung, daher versuche ich es mit einem neuen.

Ich habe mir den HmIP RF-USB Stick von ELV besorgt und diesen an einem RPi 4 eingebunden und versuche diesen nun im RaspberryMatic Add-On von Home Assistant einzubinden.

lsusb zeigt den USB-Stick an.
detect_radio_module liefert ebenfalls ein positives Ergebnis, dass der Stick wohl auch funkt.
In der Hilfe-Seite von RaspberryMatic, sehe ich auch, dass die Software das Modul wohl erkannt hat.
Irgendwo habe ich gelesen, dass man den Stick selbst als "Gerät anlernen" muss, aber hier kommt nun das Problem, RM zeigt nun die Meldung "Es wurde keinerlei nutzbares HM Funkmodul....gefunden"

Ich habe das halbe Netz und das Forum abgegrast, um mein Problem zu finden, aber alle Maßnahmen führten nicht zum Erfolg.

Noch ein paar Eckdaten: 
das Funkmodul hat wohl die Firmware 4.4.18
das RaspberryMatic Addon ist in der Version 3.65.8.20220831
Kernelversion des Raspbian (bullseye) ist 5.15.61-v7l+

Eine Auffälligkeit, die ich entdeckt hatte, dass mir die Infoseite zeigt, das Modul sei mit /dev/raw-uart verknüpft, detect_radio_module konnte es aber nur auf /dev/mmd_hmip ansprechen. 

Ich hoffe, dass mir jemand weiterhelfen kann. Vielen Dank schon im Voraus dafür.

**Comment by jens-maus** [comment]
> das RaspberryMatic Addon ist in der Version 3.65.8.20220831
> Kernelversion des Raspbian (bullseye) ist 5.15.61-v7l+

Aus diesen Daten schließe ich (das hattest du nicht aufgelistet), das du wohl eine "HomeAssistant Supervised" Installation unter einem RaspberryPiOS bzw. Standard-Linux verwendest und nicht HomeAssistantOS einsetzt. Und auch wenn das mit "HomeAssistantz Supervised" prinzipiell funktioniert, so ist dies gerade im Bezug auf RaspberryMatic bzw. direkte kernel modul zugriffe so eine Sache. Also ist das so? Hast du Home Assistant als Supervised Variante unter einem Standard Raspbian installiert, ja?

Wenn dem nämlich so ist, wirst du vmtl. wie viele andere vor dir in die gewissen Probleme rennen die das HomeAssistantOS selbst mit viel KnowHow und gewissen patches an CGroup&Co versucht zu umgehen damit von innerhalb eines Addon Docker Containers trotzdem direkte Hardwarezugriffe möglich werden.

Genau deshalb gibt es ja einen gesonderten Eintrag in der Dokumentation zu "Home Assistant Supervised" Installationen. Siehe: https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#using-homeassistant-supervised Hier gitl vor allem die **Warning** ganz unten zu beachten.

Ansonsten bitte einfach hier mal die Ausgaben des RaspberryMatic Startlogs (`/var/log/boot.log`) zeigen, sowie am besten einen Screenshot der Hilfeseite der RaspberryMatic WebUI in der alle notwendigen Zusatzinfos über Hardware+Funkmodul zu sehen sind.

**Reply by cfkone** [reply]
Danke für die schnelle Antwort.

Ja, die Vermutung stimmt, dass ich die Supervised-Installation gewählt habe. Auf einem bereits bestehendem Raspi-System, auf dem auch andere Dienste bereits laufen. Daher auch die Entscheidung. Hintergrund ist, dass ich eine "reine" isolierte RaspberryMatic-Installation auf einem RPi 3 (mit GPIO-Modul) ablösen will und gleichzeitig die Möglichkeiten von Home Assistant nutzen wollte (da noch andere SmartHome-Systeme vorhanden sind)

Dass Docker (logischerweise!) das Problem sein könnte, ist mir so noch gar nicht in den Sinn gekommen. Die o.g. Anleitung bin ich in der Tat 1:1 so durchgegangen. Die Warning habe ich allerdings nicht realisiert. Das sollte ich mir nochmal anschauen.

Das Startlog ist nicht unter /var/log zu finden, sondern kann ich nur(?) im Add-On in HA abgreifen
![image](https://user-images.githubusercontent.com/1564467/188859261-2b0da31b-d14d-4819-8508-1214a55d6e84.png)

Der Screenshot der RM WebUI 
![image](https://user-images.githubusercontent.com/1564467/188859486-35e59850-f0ca-4cf8-8cd4-2166a9280d59.png)

Vielen Dank schon mal soweit.
(PS: womöglich sollte ich es doch überdenken und das HA-OS nehmen und dafür auch eine separate Maschine!?)

**Reply by jens-maus** [reply]
> Ja, die Vermutung stimmt, dass ich die Supervised-Installation gewählt habe. 

Also zu meiner Überraschung sieht dein Startlog und auch die Ausgaben der Hilfeseite auf der WebUI eigentlich recht gut aus. Denn wenn es dieses "HA Supervised" problem ist, dann sollte hier auch kein auslesen des Funkmodules möglich sein. Sollte also eigentlich funktionieren, denn der HmIP-RFUSB wird ja anscheinend richtig angebunden undsollte somit auch innerhalb der WebUI problemlos funktionieren. Wenn du also in den "Posteingang" wechselst sollte doch der HmIP-RFUSB auftauchen und dann unter "Einstellungen -> Geräte" auch normal aufgelistet werden. Zumindest kann ich aus dem Log+Screenshot nicht erkennen warum das nicht der Fall sein sollte. Was du natürlich noch probieren kannst ist, nochmal ein kompletten System Reset innerhalb der WebUI zu machen, dann das Addon nochmal hochfahren zu lassen und dann sollte das RaspberryMatic Add-on eigentlich komplett funktionieren wie eine reelle RaspberryMatic.

Achja, und in deinem Logoutput fehlen aber noch ein par zeilen ganz unten, denn er müsste eigentlich den "rfd" und "HMIPServer" starten, das kann ich aber aus dem Screenshot nicht sehen weil die letzten Zeilen dort "Starting hs485d..." ist.

> (PS: womöglich sollte ich es doch überdenken und das HA-OS nehmen und dafür auch eine separate Maschine!?)

Das kannst du natürlich auch gerne probieren und wäre IMHO die sauberste Variante. Denn selbst wenn du das mit HA Supervised nun hinbekommst, so wird es vmtl. nicht lange dauern bis ein RaspberryPiOS Update rauskommt das wieder funktionalitäten dann innerhalb des virtuellen HomeAssistant dann kaputt macht. Das RaspberryMatic Add-on sollte unter HomeAssistantOS wirklich out-of-the-box funktionieren.

**Comment by cfkone** [comment]
Ja, das ist auch der Grund für meine kleine "Verzweiflung", da eben keine nennenswerter Fehler zu erkennen ist. Aber womöglich habe ich die Stelle noch nicht gefunden.

```
Tatsächlich fehlten im Log noch diese Zeilen
Starting multimacd: ...........ERROR
Starting rfd: ERROR: /dev/mmd_bidcos missing, no BidCos-RF hardware found
Starting HMIPServer: ERROR: /dev/mmd_hmip missing
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
find: /usr/local/etc/monit*.cfg: No such file or directory
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.65.8.20220831 (raspmatic_oci_arm)
```

Und hier ist wohl doch noch ein Problem mit dem HMIPServer bzw. rfd. Ist das womöglich die Ursache??

das device /dev/mmd_hmip ist im Hostsystem ja tatsächlich mit dem USB-Stick belegt. 
D.h. dass der HA-Docker-Container wohl das Device nicht erreichen kann?

**Reply by jens-maus** [reply]
> Starting multimacd: ...........ERROR
> Starting rfd: ERROR: /dev/mmd_bidcos missing, no BidCos-RF hardware found
> Starting HMIPServer: ERROR: /dev/mmd_hmip missing

Na da haben wir doch das problem. der `multimacd` kann sich nicht starten weil er wohl die `/dev/mmd_bidcos` und `/dev/mmd_hmip` devices des `eq3_char_loop` kernel modules nicht dynamisch anlegen kann. Dieses kernel modul lädt das Add-on eigentlich dynamisch beim starten nach und der multimacd legt dann diese devices dynamisch an. Und genau hier liegt das Problem. der `multimacd` verwendet hierfür zugriffe die im Host-OS besonderes augemerk bzw. zugriffsrechte brauchen wenn diese aus einem Docker Container heraus passieren (Stichwort CGroups). Geh also wie am Anfang beschrieben nochmal die Anleitung genau durch und nimm den **Warning** Hinweis entsprechend ernst. Ich kann mir gut vorstellen das das bereits die Lösung ist und du einfach das gleiche `CGroups` Problem hast wie dort beschrieben.

Wenn das allerdings die Lösung ist, sollte dir hoffentlich trotzdem klar sein das du hier eine sehr komplexe Art genutzt hast RaspberryMatic zu installieren. Ich würde daher immer noch den Umstieg auf HomeAssistantOS+RaspberryMatic Add-on raten.

**Reply by cfkone** [reply]
Okay, dann habe ich zumindest das Problem verstanden und versuche da mal tiefer zu bohren. Werde aber den Switch zu HA-OS in Betracht ziehen..

Auf jeden Fall danke für die schnelle und kompetente Unterstützung. Ich werde meine letztendlichen Ergebnisse gerne nochmal hier posten.

**Reply by cfkone** [reply]
Abschließend die Info:

tatsächlich hat die Installation auf einem Home Assistant OS und dem Addon prompt funktioniert. Der USB Stick wird (wie erwartet) direkt als HM-Zentrale im Posteingang aufgelistet. 
Ich (selbst Softwareentwickler) fühle mich zwar durch die "Probleme" mit HA Supervised herausgefordert, weiß aber nicht, ob ich die Mühe da noch investieren mag.

Auf jeden Fall nochmal Danke für die Hilfe.
